### PR TITLE
wolfi-packages: add gke-gcloud-auth-plugin

### PR DESCRIPTION
**Pull request by michaellzc**
for https://github.com/sourcegraph/sourcegraph/pull/56426 and https://github.com/sourcegraph/controller/pull/1040

follow https://docs.sourcegraph.com/dev/how-to/wolfi/add_update_packages#create-a-new-package

## Test plan

CI

tested locally

```
docker run -it --entrypoint /bin/sh sourcegraph-wolfi/cloud-mi2-base:latest-amd64
```

```sh
/ # terraform version
Terraform v1.5.6
on linux_amd64

/ # gke-gcloud-auth-plugin --version
Kubernetes v0.0.0-master+$Format:%H$

/ # ls /usr/bin | grep gcloud
gke-gcloud-auth-plugin
/ # 
```

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #56438** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56438" target="_blank">Graphite</a>  👈
    * **PR #56426** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56426" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56438?utm_source=stack-comment).

File: wolfi-packages/terraform.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# - bump version or epoch
+# - update `.pipeline[0].with.expected-commit` to the coresponding commit sha
+#   you can find it from https://github.com/hashicorp/terraform/releases
+#
+# changelog
+# - 2023-09-07: bump to 1.5.6
+
+package:
+  name: terraform
+  version: 1.5.6
```
Comment (willdollman) [inline wolfi-packages/terraform.yaml:15]: Is this change permitted by the new terraform license? I can't find an exact page confirming version, but https://www.hashicorp.com/license-faq#aug-10-announcement states all release after August 10th are BSL-licensed, and 1.5.6 was released August 23rd.

Comment (michaellzc) [reply wolfi-packages/terraform.yaml:15]: ughh fuck, maybe that's why they disable the auto update for all hashi stuff https://github.com/wolfi-dev/os/commit/afeab0d2f7e60781477361a29a2ec42888c5a578

Comment (michaellzc) [reply wolfi-packages/terraform.yaml:15]: not a lawyer, but if we're not competing with hashi in anyway, and we're not shipping this to customer

i doubt we will be violating the term os use

File: wolfi-packages/gke-gcloud-auth-plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  - uses: go/build
+    with:
+      packages: ./cmd/gke-gcloud-auth-plugin
+      output: gke-gcloud-auth-plugin
+      # TODO(mattmoor): Consider adding all of these:
+      # https://github.com/kubernetes/cloud-provider-gcp/blob/e64cf3b0fcb1958cee1fe55d7e30f4573c34bf4e/defs/version.bzl#L38
+      ldflags: -s -w
+
+  - uses: strip
```
Comment (willdollman) [inline wolfi-packages/gke-gcloud-auth-plugin.yaml:44]: You can add back the `update:` block that was at the bottom of the original file you forked from. We don't use this section currently, but we can in the future to detect when there are upstream fixes.

File: wolfi-packages/terraform.yaml (hunk 1 above)
Comment (willdollman) [reply wolfi-packages/terraform.yaml:15]: Additional context on the wolfi side: https://www.chainguard.dev/unchained/chainguard-images-users-on-hashicorp-license-changes

Comment (michaellzc) [reply wolfi-packages/terraform.yaml]: extracted to a separate PR https://github.com/sourcegraph/sourcegraph/pull/56441 to unblock this first

**Comment by michaellzc** [comment]
merged into https://github.com/sourcegraph/sourcegraph/pull/56426

**Pull request closed without merging**